js/documents.js

```diff
@@ -487,7 +485,7 @@ var documentsMain = {
 			    "&lang=" + OC.getLocale().replace('_', '-') +
 			    "&closebutton=1" +
 			    "&revisionhistory=1";
-			if (!documentsMain.canEdit || action === "view") {
+			if (action === "view") {
```

Comment:
this and other are really not needed nor used. actually "action" should determine how document should be opened. Then token is created with proper permissions, and if these are missing in the token worst case document instead of in edit opens in view mode.